models/user/must_change_password.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package user
+
+import (
+	"context"
+	"strings"
+
+	"code.gitea.io/gitea/models/db"
+	"code.gitea.io/gitea/modules/util"
+
+	"xorm.io/builder"
+)
+
+func SetMustChangePassword(ctx context.Context, all, mustChangePassword bool, include, exclude []string) (int64, error) {
+	toInterfaceSlice := func(input []string) []interface{} {
+		output := make([]interface{}, 0, len(input))
+		for _, in := range input {
+			in = strings.ToLower(strings.TrimSpace(in))
+			if in == "" {
+				continue
+			}
+			output = append(output, in)
+		}
+		return output
+	}
+
+	var cond builder.Cond
+
+	// Only include the users where something changes to get an accurate count
+	cond = builder.Neq{"must_change_password": mustChangePassword}
+
+	if !all {
+		toInclude := toInterfaceSlice(include)
+		if len(toInclude) == 0 {
+			return 0, util.NewSilentWrapErrorf(util.ErrInvalidArgument, "no users to include provided")
+		}
+
+		cond = cond.And(builder.In("lower_name", toInclude...))
```

Comment:
[]string is acceptable, so I don't think this conversation from `[]string` to `[]interface` is necessary. https://gitea.com/xorm/builder/src/branch/master/cond_in.go#L216

Reply:
If that does work I think it would be better, it cleans up the code above a bit.

Reply:
```
# code.gitea.io/gitea/models/user
models/user/must_change_password.go:40:44: cannot use toInclude (variable of type []string) as []interface{} value in argument to builder.In
models/user/must_change_password.go:45:47: cannot use toExclude (variable of type []string) as []interface{} value in argument to builder.NotIn
```

Reply:
🤦‍♂️ didn't notice `cond.In` vs `builder.In`

Reply:
We're still going to need the trimspace remove empty function, but I'll do it.

Reply:
Ah shoot fair enough, I was hoping we would be able to remove the inline func, but I hadn't noticed the string trimming as well. 

I guess either way LGTM then. Sorry for the noise.